ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/KeyValueCommitting.scala

```diff
@@ -73,13 +73,22 @@ object KeyValueCommitting {
     try {
       submission.getPayloadCase match {
         case DamlSubmission.PayloadCase.PACKAGE_UPLOAD_ENTRY =>
-          ProcessPackageUpload(
+          val (logEntry, outputs) = PackageCommitter(engine).run(
```

Comment:
further step would be to initialize all these "committers" once. Needed to nicely set up some metrics' gauges anyway, but that'll change the interface so would do this in multiple steps (and would try to keep the original interface around until implementations have upgraded).